src/GovUk.Education.ExploreEducationStatistics.Common/Cache/CacheAspect.cs

```diff
@@ -349,7 +349,8 @@ private static bool HasUnambiguousMatch(List<ParameterInfoScore> matches)
             {
                 score += 1;
             }
-            else if (constructorParam.Name == cacheKeyParamInfo.Name)
+            else if (constructorParam.Name is not null 
+                     && constructorParam.Name.Equals(cacheKeyParamInfo.Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
Why ignore case? Wouldn't this be a problem in cases where casing might matter?

Can't think of anything off the top of my head, but it feels like this could potentially be a bit of a bug. Variables in general are treated case sensitively so this doesn't sit right.

Same question for the change below as well.

Reply:
So the reason behind this change is to allow us to use records and their default constructors as CacheKeys, whereas prior to this we've had to use classes.  So we can specify a nice concise key like:

```
public record GetReleaseCacheKey(string PublicationSlug, string ReleaseSlug) : IMemoryCacheKey
{
    public string Key => $"{GetType().Name}:{PublicationSlug}/{ReleaseSlug}";
}
```

rather than its more verbose class equivalent, or a record but with an explicit `public GetReleaseCacheKey(string publicationSlug, string releaseSlug) {...}` constructor.

The need for the case insensitivity is down to the case convention differences in the record's default constructor (PascalCase) versus the explicit / class constructor (camelCase).

As we're taking method params from a Controller / Service / Whatever method (which by convention will be camelCase) and attempting to map them to a default record constructor (PascalCase), this is the driver for this.